CIP-0118/README.md

```diff
@@ -0,0 +1,466 @@
+to accept the non-primary asset offer in exchange for paying the transaction fee. This is the core
+requirement of Babel fees. In addition, the goal of this design is to lay the groundwork for
+supporting more general intents and the possibility of extending to additional ones.
+We have done so by adding the following set of features :
+
+1. transactions can be batched
+- batch contains one top-level transaction, with multiple possible sub-transactions ;
+
+2. the batch must be balanced, but individual transactions in it do not have to be, so
+- transactions in a single batch may be used to pay parts of the fee and satisfy `minUTxOValue` requirement for other sub-transactions;
+- unbalanced transactions can be interpreted as swap offers, getting resolved within a complete batch ;
+
+3. script data is shared across all transactions in a single batch
+- enables script deduplication ;
+
+4. batch script guards enable sub-transactions to specify properties required of batches in which they may be included
+- represents all possible intents that do not require ledger rule relaxation ;
+
+5. the top-level transaction must provide collateral for all scripts in its sub-transactions
+- sub-transactions are neither required nor allowed to cover their own collateral ;
+
+
+## Use cases
+
+### Open (atomic) swaps
+
+A user wants to swap 10 Ada for 5 tokens `myT`. He creates an unbalanced transaction `tx` that
+has extra 10 Ada, but is short 5 `myT`.
+Any counterparty that sees this transaction can create a top-level
+transaction `tx'` that includes `tx` as a sub-transaction. The transaction
+`tx'` would have extra 5 `myT`, and be short 10 Ada.
+
+### DEX aggregators
+
+A DEX aggregator running a Babel service would set its filter to only accept transactions
+trading tokens it is interested in, and are being traded at a favourable price.
+It may or may not require incoming transactions to cover their own fees and/or collateral.
+In the case a set of sub-transactions is not fully balanced, and the aggregator is interested
+in participating in the exchange offers within the batch, it would have to include its own
+liquidity pool as one of the inputs
+in the top-level transaction, and output any extra tokens to its updated liquidity pool UTxO.
+
+### Babel fees
+
+Babel fees is a specific subtype of the first usecase where the missing assets are necessarily a
+quantity of Ada. Usually, it also implies that no Ada is contained in the inputs of such a
+transaction, aside from the minimal amount enforced by `minUTxOValue` requirement, and the missing
+Ada goes towards paying transaction fees.
+
+### DApp fee and min-UTxO sponsorship
+
+DApp may run a Babel service which only cares only about transactions interacting with a
+certain DApp, discarding all other incoming transactions immediately. This service
+may include calculating ExUnits, paying fees/minUTxO for, and providing collateral for such
+partially valid transactions.
+
+## Pseudo-code Example
```

Comment:
Likewise we have no top-level section by this name, so unless moving to the end in an _Appendix_ this would be part of the _Specification_ (OK since the "use cases" should probably be introduced in the _Motivation_ section immediately before the spec).